java-checks-test-sources/src/main/java/checks/StringConcatToTextBlockCheck.java

```diff
@@ -0,0 +1,66 @@
+package checks;
+
+public class StringConcatToTextBlockCheck {
+  
+  private static final int SOME_INT = 5;
+  
+  public static void main(String[] args) {
+
+    String s = "<html>\n" + // Noncompliant [[sc=16;ec=25]] {{Replace this String concatenation with Text block.}}
+               "    <body>\n" +
+               "        <tag>\n" +
+               "        </tag>\n" +
+               "    </body>\n" +
+               "</html>";
```

Comment:
What's a good example of what we would like to find with this rule:

- a list of string literal concatenation
- each string literal end with \n (it is optional for the last one)

The current rule implementation is super simple, that's nice.
Unfortunately, splitting on `"\\\\n"` raises the following false-positive:
```
    System.out.println("\\n-" +
      "\\n-\\n");
```
Also, I don't like proposing a text block solution that is longer, like for:
```
    System.out.println("-\n-\n-\n-\n-" +
      "\n-\n");
```
Text bock is a longer solution:
```
    System.out.println(
      """
      -
      -
      -
      -
      -
      -
      """);
```
And some string with spaces before \n could be easier to read like:
```
    System.out.println(" \n" +
      "  \n   \n");
```
compare to:
```
    System.out.println(
      """
      \u0020
      \u0020\u0020
      \u0020\u0020\u0020
      """);
```